src/azure-cli/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -516,6 +516,10 @@ def _build_storage_profile():
             'vTpmEnabled': enable_vtpm
         }
 
+    # Compatibility of various API versions
```

Comment:
will this impact non-Azure  stack profile?

Reply:
It is an optional field. But in old API version, it is not supported. The updated usage is better.